cylc/flow/prerequisite.py

```diff
@@ -223,6 +224,45 @@ def dump(self):
         # (Else trigger wiped out by pre-initial simplification.)
         return res
 
+    def api_dump(self, workflow_id):
+        """Return list of populated Protobuf data objects."""
+        if self.conditional_expression:
+            temp = self.get_raw_conditional_expression()
+            temp = temp.replace('|', ' | ')
+            temp = temp.replace('&', ' & ')
+        else:
+            for s_msg in self.satisfied:
+                temp = self.MESSAGE_TEMPLATE % s_msg
+        conds = []
+        num_length = int(math.ceil(float(len(self.satisfied)) / float(10)))
+        for ind, message_tuple in enumerate(sorted(self.satisfied)):
+            name, point = message_tuple[0:2]
+            t_id = f"{workflow_id}/{point}/{name}"
+            char = 'c%.{0}d'.format(num_length) % ind
+            c_msg = self.MESSAGE_TEMPLATE % message_tuple
+            c_val = self.satisfied[message_tuple]
+            c_bool = bool(c_val)
+            if c_val is False:
+                c_val = "unsatisfied"
+            cond = PbCondition(
+                task_proxy=t_id,
+                expr_alias=char,
+                req_state=message_tuple[2],
+                satisfied=c_bool,
+                message=c_val,
+            )
+            conds.append(cond)
+            temp = temp.replace(c_msg, char)
+        if self.satisfied:
```

Comment:
After reading the code here and in the IDE (to locate what is referenced where), it looks to me that a) this method never alters the state of the current object, and also does not return anything unless `self.satisfied` is `True`.

Then, could we move this `if self.satisfied` to the top of the function? We are doing other `if/else`, some calculation, and a `for` loop before this check... and it looks to me like we could start the function by checking if satisfied, and only proceeding if `True`, returning immediately otherwise?